File: backend/app/schemas/simulation.py
from pydantic import BaseModel


class SimulationPercentiles(BaseModel):
    p10: float
    p50: float
    p90: float


class SimulationMetrics(BaseModel):
    expected_final_value: float
    probability_of_loss: float
    estimated_volatility: float
    max_drawdown: float
    percentiles: SimulationPercentiles


class SimulationPathPoint(BaseModel):
    step: int
    portfolio_value: float


class PortfolioSimulationResult(BaseModel):
    initial_budget: float
    num_simulations: int
    time_horizon_months: int
    metrics: SimulationMetrics
    sample_path: list[SimulationPathPoint]
    summary: str

File: frontend/src/types/simulation.ts
export type SimulationPercentiles = {
  p10: number;
  p50: number;
  p90: number;
};

export type SimulationMetrics = {
  expected_final_value: number;
  probability_of_loss: number;
  estimated_volatility: number;
  max_drawdown: number;
  percentiles: SimulationPercentiles;
};

export type SimulationPathPoint = {
  step: number;
  portfolio_value: number;
};

export type SimulationBandPoint = {
  step: number;
  p10: number;
  p50: number;
  p90: number;
};

export type PortfolioSimulationResponse = {
  initial_budget: number;
  num_simulations: number;
  time_horizon_months: number;
  metrics: SimulationMetrics;
  sample_path: SimulationPathPoint[];
  percentile_band: SimulationBandPoint[];
  summary: string;
};